apps/web/src/components/HeroUpdated/Hero.tsx

```diff
@@ -35,32 +37,39 @@ const paddingMap = new Map([
   ],
 ]);
 
+interface NavTab {
+  url: string;
+  label: string;
+}
+
 interface HeroProps {
   imgPath?: string;
   title: ReactNode;
   subtitle?: ReactNode;
   crumbs?: BreadcrumbsProps["crumbs"];
-  simpleCrumbs?: boolean;
+  buttonLinks?: ButtonLinkType[];
+  navTabs?: NavTab[];
   children?: ReactNode;
   centered?: boolean;
-  linkSlot?: ReactNode;
+  overlap?: boolean; // only takes effect if navTabs is also undefined
```

Comment:
Hmm, I wonder if we can force a compile-time check of this with some clever TypeScript?  I'm imagining we have an  "abstract HeroProps" interface like what you have here and then two "actual HeroProps" interfaces that extend it where one has the overlap and no navTabs and the other has no overlap but does have navTabs.  Then have the component expect actual 1 or actual 2.  Let me know if you want to chat about this.

Reply:
Not sure if it is even worth fretting about it? I think the comment should be self-explanatory and all that happens is the overlap gets ignored while nothing breaks.

Reply:
I always want to move runtime errors to compile-time errors!  :grin: 

Uh oh, the current logic doesn't seem to work.
![image](https://github.com/user-attachments/assets/8e130afc-8cf5-4cae-8453-3007a78ad779)

Here's what I tried:
```tsx
export const ConflictingProps = TemplateWithContent.bind({});
ConflictingProps.args = {
  overlap: true,
  navTabs: [
    { url: "#one", label: "Tab One" },
    { url: "#two", label: "Tab Two" },
    { url: "#three", label: "Tab Three" },
    { url: "#four", label: "Tab Four" },
    { url: "#five", label: "Tab Five" },
  ],
  children: <FormComponent></FormComponent>,
};
```

Reply:
I took this as an interesting TypeScript challenge and learning experience for myself.  I think I got it working!

```tsx
interface HeroSharedProps {
  imgPath?: string;
  title: ReactNode;
  subtitle?: ReactNode;
  crumbs?: BreadcrumbsProps["crumbs"];
  buttonLinks?: ButtonLinkType[];
  children?: ReactNode;
  centered?: boolean;
}

type HeroWithNavTabsProps = HeroSharedProps & {
  navTabs?: NavTab[];
  overlap?: never;
};

type HeroWithOverlapProps = HeroSharedProps & {
  navTabs?: never;
  overlap: boolean;
};

const Hero = (props: HeroWithNavTabsProps | HeroWithOverlapProps) => {
  // shared props
  const {
    imgPath,
    title,
    subtitle,
    crumbs,
    buttonLinks,
    children,
    centered = false,
  } = props;

  // conditional props
  const navTabs = "navTabs" in props ? props.navTabs : null;
  const overlap = "overlap" in props ? props.overlap : false;
```

Reply:
Thank you for sketching that out for me! I have included it in https://github.com/GCTC-NTGC/gc-digital-talent/pull/11713/commits/5b4605ff99949e188a9bbacfc2a528d84a74bbd8